just-premium-htb.js

```diff
@@ -66,10 +66,170 @@ function JustPremiumHtb(configs) {
+                    var len = bids[zoneId].length;
+                    while (len--) {
+                        if (__passCond(params, bids[zoneId][len])) {
+                            return bids[zoneId].splice(len, 1).pop();
+                        }
+                    }
+                }
+            }
+        }
+
+        return false;
+    }
+
+    function __passCond(params, bid) {
+        var format = bid.format;
+
+        if (params.allow && params.allow.length) {
+            return params.allow.indexOf(format) > -1;
+        }
+
+        if (params.exclude && params.exclude.length) {
+            return params.exclude.indexOf(format) < 0;
+        }
+
+        return true;
+    }
+
+    function __arrayUnique(array) {
+        const a = array.concat();
+        for (var i = 0; i < a.length; ++i) {
+            for (var j = i + 1; j < a.length; ++j) {
+                if (a[i] === a[j]) {
+                    a.splice(j--, 1);
+                }
+            }
+        }
+
+        return a;
+    }
+
+    function __preparePubCond(bids) {
+        const cond = {};
+        const count = {};
+
+        bids.forEach(function (bid) {
+            var zone = bid.zoneId;
+
+            if (cond[zone] === 1) {
+                return;
+            }
+
+            const allow = bid.allow || [];
+            const exclude = bid.exclude || [];
+
+            if (allow.length === 0 && exclude.length === 0) {
+                return cond[zone] = 1;
+            }
+
+            cond[zone] = cond[zone] || [[], {}];
+            cond[zone][0] = __arrayUnique(cond[zone][0].concat(allow));
+            exclude.forEach(function (e) {
+                if (!cond[zone][1][e]) {
+                    cond[zone][1][e] = 1;
+                } else cond[zone][1][e]++;
+            });
+            count[zone] = count[zone] || 0;
+            if (exclude.length) count[zone]++;
+        });
+
+        Object.keys(count).forEach(function (zone) {
+            if (cond[zone] === 1) {
+                return;
+            }
+
+            const exclude = [];
+            Object.keys(cond[zone][1]).forEach(function (format) {
+                if (cond[zone][1][format] === count[zone])
+                    exclude.push(format);
+            });
+            cond[zone][1] = exclude;
+        });
+
+        Object.keys(cond).forEach(function (zone) {
+            if (cond[zone] !== 1 && cond[zone][1].length) {
+                cond[zone][0].forEach(function (r) {
+                    var idx = cond[zone][1].indexOf(r);
+                    if (idx > -1) {
+                        cond[zone][1].splice(idx, 1);
+                    }
+                })
+                ;
+                cond[zone][0].length = 0;
+            }
+
+            if (cond[zone] !== 1 && !cond[zone][0].length && !cond[zone][1].length) cond[zone] = 1;
+        });
+
+        return cond;
+    }
+
+    function __requestResource(tagSrc) {
```

Comment:
We recommend you to avoid requesting for external resources as it introduces additional latency. Since the wrapper have a global time out, this additional latency to your bidder may decrease your chance of making the auction. Please let us know if you are open to not having external resources.

Reply:
@ix-certification we are working to remove it but for the time being we would like to keep this in place as without it we would not be able to show the ad.

Reply:
Ok, we'll take this into consideration when doing end-to-end testing and verify performance